aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -3769,3 +3848,62 @@ func resourceAviatrixTransitGatewayDelete(d *schema.ResourceData, meta interface
 
 	return nil
 }
+
+// count the number of WAN interfaces in the interface config
+func countInterfaceTypes(interfaceList []interface{}) (int, error) {
+	wanCount := 0
+	// Iterate over each interface in the list
+	for _, item := range interfaceList {
+		interfaceMap, ok := item.(map[string]interface{})
+		if !ok {
+			return 0, fmt.Errorf("invalid interface entry, expected a map")
+		}
+		interfaceType, ok := interfaceMap["type"].(string)
+		if !ok {
+			return 0, fmt.Errorf("interface type must be a string")
+		}
+		// Count WAN interfaces
+		if interfaceType == "WAN" {
+			wanCount++
+		}
+	}
+	return wanCount, nil
+}
+
+// get the interface name (ethX) from interface type, index, and counts
+func getInterfaceName(intfType string, intfIndex, wanCount int) (string, error) {
+	// Check if the interface type is valid
+	if intfType != "WAN" && intfType != "MANAGEMENT" {
+		log.Printf("Invalid interface type %s", intfType)
+		return "", errors.New("invalid interface type " + intfType)
+	}
+	var num int
+	// Determine the interface name based on the type and index
+	switch intfType {
+	case "WAN":
```

Comment:
I'm a little bit concerned about the interface mapping here:
With wanCount = 2 and intfIndex = 2 for both WAN and MANAGEMENT interfaces, both interfaces will map to num = 3, which corresponds to eth3. I don't know if that particular use case is valid or will be valid in the future. 

I would say regardless at some point (doesn't need to be this PR) we need to document the logic for interface mapping for what is and is not valid, so we can compare the code here to the actual intentions.  This seem brittle to me.

Reply:
The interface mappings are validated in the backend api call. If the mgmt interface or wan interface in the above case is not set then the api throws an error.